### PR TITLE
Update 4-term parameter shift coefficients

### PR DESCRIPTION
**Pull request by josh146**
**Description of the Change:** Updates the four-term parameter-shift rule to use coefficients that minimize the variance as per https://arxiv.org/abs/2104.05695

**Benefits:** More optimal choice of coefficients

**Possible Drawbacks:** n/a

**Related GitHub Issues:** n/a

### CODE REVIEW

File: pennylane/ops/qubit.py [hunk 1]
```diff
@@ -1230,8 +1230,18 @@ class CRX(Operation):
...
     * Number of wires: 2
     * Number of parameters: 1
-    * Gradient recipe: :math:`\frac{d}{d\phi}f(CR_x(\phi)) = \frac{1}{2}\left[f(CR_x(\phi+\pi/2)) - f(CR_x(\phi-\pi/2))\right]`
-      where :math:`f` is an expectation value depending on :math:`CR_x(\phi)`.
+    * Gradient recipe: The controlled-RX operator satisfies a four-term parameter-shift rule
+      (see Appendix F, https://arxiv.org/abs/2104.05695):
+
+      .. math::
+
+          \frac{d}{d\phi}f(CR_x(\phi)) = c_+ \left[f(CR_x(\phi+a)) - f(CR_x(\phi-a))\right] + c_- \left[f(CR_x(\phi-b)) - f(CR_x(\phi+b))\right]
```
Comment (antalszava) [inline pennylane/ops/qubit.py:1238]: Probably personal preference, but how about adjusting the signs as per eq. (F14)? :thinking:
```suggestion
          \frac{d}{d\phi}f(CR_x(\phi)) = c_+ \left[f(CR_x(\phi+a)) - f(CR_x(\phi-a))\right] - c_- \left[f(CR_x(\phi+b)) - f(CR_x(\phi-b))\right]
```

File: pennylane/ops/qubit.py [hunk 2]
```diff
@@ -1419,9 +1449,19 @@ class CRot(Operation):
...
     * Number of parameters: 3
-    * Gradient recipe: :math:`\frac{d}{d\phi}f(CR(\phi, \theta, \omega)) = \frac{1}{2}\left[f(CR(\phi+\pi/2, \theta, \omega)) - f(CR(\phi-\pi/2, \theta, \omega))\right]`
-      where :math:`f` is an expectation value depending on :math:`CR(\phi, \theta, \omega)`.
-      This gradient recipe applies for each angle argument :math:`\{\phi, \theta, \omega\}`.
+    * Gradient recipe: The controlled-Rot operator satisfies a four-term parameter-shift rule
+      (see Appendix F, https://arxiv.org/abs/2104.05695):
+
+      .. math::
+
+          \frac{d}{dx_i}f(CR(\mathbf{x}_i)) = c_+ \left[f(CR(\mathbf{x}_i+a)) - f(CR(\mathbf{x}_i-a))\right] + c_- \left[f(CR(\mathbf{x}_i-b)) - f(CR(\mathbf{x}_i+b))\right]
```
Comment (antalszava) [inline pennylane/ops/qubit.py:1457]: Would we like to make `x` bold there too?
```suggestion
          \frac{d}{d\mathbf{x}_i}f(CR(\mathbf{x}_i)) = c_+ \left[f(CR(\mathbf{x}_i+a)) - f(CR(\mathbf{x}_i-a))\right] + c_- \left[f(CR(\mathbf{x}_i-b)) - f(CR(\mathbf{x}_i+b))\right]
```

File: pennylane/ops/qubit.py [hunk 3]
```diff
@@ -1419,9 +1449,19 @@ class CRot(Operation):
...
+
+      where :math:`f` is an expectation value depending on :math:`CR(\mathbf{x}_i)`, and
+
+      - :math:`\mathbf{x} = (\phi, \theta, \omega)`
```
Comment (antalszava) [inline pennylane/ops/qubit.py:1461]: `i` doesn't seem to be defined
```suggestion
      - :math:`\mathbf{x} = (\phi, \theta, \omega)` and `i` is an index to :math:`\mathbf{x}`
```

**Review by antalszava — APPROVED** [review]
Looks good to me! :100: :confetti_ball: None of the suggestions are blockers.

File: pennylane/ops/qubit.py (hunk 1 above)
Comment (josh146) [reply pennylane/ops/qubit.py:1238]: :+1:

**Comment by dwierichs** [comment]
This looks good to me, up to a tiny documentation issue: The latest change of the sign in the CRX docstring would need to be applied to CRY, CRZ and CRot as well, wouldn't it?

File: pennylane/ops/qubit.py [hunk 4]
```diff
@@ -1289,8 +1299,18 @@ class CRY(Operation):
...
     * Number of wires: 2
     * Number of parameters: 1
-    * Gradient recipe: :math:`\frac{d}{d\phi}f(CR_y(\phi)) = \frac{1}{2}\left[f(CR_y(\phi+\pi/2)) - f(CR_y(\phi-\pi/2))\right]`
-      where :math:`f` is an expectation value depending on :math:`CR_y(\phi)`.
+    * Gradient recipe: The controlled-RY operator satisfies a four-term parameter-shift rule
+      (see Appendix F, https://arxiv.org/abs/2104.05695):
+
+      .. math::
+
+          \frac{d}{d\phi}f(CR_y(\phi)) = c_+ \left[f(CR_y(\phi+a)) - f(CR_y(\phi-a))\right] + c_- \left[f(CR_y(\phi-b)) - f(CR_y(\phi+b))\right]
```
Comment (josh146) [inline pennylane/ops/qubit.py:1307]:
```suggestion
          \frac{d}{d\phi}f(CR_y(\phi)) = c_+ \left[f(CR_y(\phi+a)) - f(CR_y(\phi-a))\right] - c_- \left[f(CR_y(\phi+b)) - f(CR_y(\phi-b))\right]
```

File: pennylane/ops/qubit.py [hunk 5]
```diff
@@ -1349,8 +1369,18 @@ class CRZ(DiagonalOperation):
...
     * Number of wires: 2
     * Number of parameters: 1
-    * Gradient recipe: :math:`\frac{d}{d\phi}f(CR_z(\phi)) = \frac{1}{2}\left[f(CR_z(\phi+\pi/2)) - f(CR_z(\phi-\pi/2))\right]`
-      where :math:`f` is an expectation value depending on :math:`CR_z(\phi)`.
+    * Gradient recipe: The controlled-RZ operator satisfies a four-term parameter-shift rule
+      (see Appendix F, https://arxiv.org/abs/2104.05695):
+
+      .. math::
+
+          \frac{d}{d\phi}f(CR_z(\phi)) = c_+ \left[f(CR_z(\phi+a)) - f(CR_z(\phi-a))\right] + c_- \left[f(CR_z(\phi-b)) - f(CR_z(\phi+b))\right]
```
Comment (josh146) [inline pennylane/ops/qubit.py:1377]:
```suggestion
          \frac{d}{d\phi}f(CR_z(\phi)) = c_+ \left[f(CR_z(\phi+a)) - f(CR_z(\phi-a))\right] - c_- \left[f(CR_z(\phi+b)) - f(CR_z(\phi-b))\right]
```

File: pennylane/ops/qubit.py [hunk 6]
```diff
@@ -1419,9 +1449,19 @@ class CRot(Operation):
...
     * Number of parameters: 3
-    * Gradient recipe: :math:`\frac{d}{d\phi}f(CR(\phi, \theta, \omega)) = \frac{1}{2}\left[f(CR(\phi+\pi/2, \theta, \omega)) - f(CR(\phi-\pi/2, \theta, \omega))\right]`
-      where :math:`f` is an expectation value depending on :math:`CR(\phi, \theta, \omega)`.
-      This gradient recipe applies for each angle argument :math:`\{\phi, \theta, \omega\}`.
+    * Gradient recipe: The controlled-Rot operator satisfies a four-term parameter-shift rule
+      (see Appendix F, https://arxiv.org/abs/2104.05695):
+
+      .. math::
+
+          \frac{d}{d\mathbf{x}_i}f(CR(\mathbf{x}_i)) = c_+ \left[f(CR(\mathbf{x}_i+a)) - f(CR(\mathbf{x}_i-a))\right] + c_- \left[f(CR(\mathbf{x}_i-b)) - f(CR(\mathbf{x}_i+b))\right]
```
Comment (josh146) [inline pennylane/ops/qubit.py:1457]:
```suggestion
          \frac{d}{d\mathbf{x}_i}f(CR(\mathbf{x}_i)) = c_+ \left[f(CR(\mathbf{x}_i+a)) - f(CR(\mathbf{x}_i-a))\right] - c_- \left[f(CR(\mathbf{x}_i+b)) - f(CR(\mathbf{x}_i-b))\right]
```

**Comment by josh146** [comment]
Thanks @dwierichs, have just made them all consistent :)

**Comment by dwierichs** [comment]
Great, thanks for carrying out this PR! :)

**Pull request merged**